main/plugins/org.talend.designer.components.localprovider/components/tFTPPut/tFTPPut_begin.javajet

```
@@ -184,14 +185,36 @@ int nb_file_<%=cid%> = 0;
 		<%if(isLog4jEnabled){%>
 			log.info("<%=cid%> - Connect to '" + <%=host %> + " has succeeded." );
 		<%}%>
-    	com.jcraft.jsch.ChannelSftp c_<%=cid%>=(com.jcraft.jsch.ChannelSftp)channel_<%=cid%>;
-    	
-    	<%if(use_encoding) {%>
+		com.jcraft.jsch.ChannelSftp c_<%=cid%>=(com.jcraft.jsch.ChannelSftp)channel_<%=cid%>;
+
+		<%if(use_encoding) {%>
 			c_<%=cid%>.setFilenameEncoding(<%=encoding%>);
 		<%}%>
 		
 	<%}%>
-	// becasue there is not the same method in JSch class as FTPClient class, define a list here
+
+<%
+	if (createDirIfNotExist) {
+%>
+		boolean dirExist_<%=cid %> = false;
+		String remotedir_<%=cid %> = <%=remotedir %>;
+		com.jcraft.jsch.SftpATTRS attrs_<%=cid %> = null;
+		try {
+			attrs_<%=cid %> = c_<%=cid%>.stat(remotedir_<%=cid %>);
+		} catch (Exception e) {
+		}
```

Need to rework empty catch block